pkg/e2e/verify/namespace_webhook.go

```diff
@@ -159,6 +182,80 @@ var _ = ginkgo.Describe(namespaceWebhookTestName, func() {
+			}
+		}, viper.GetFloat64(config.Tests.PollingTimeout))
+
+		util.GinkgoIt("Non-privileged users cannot create prometheusrule in privileged namespaces", func() {
+			for privilegedNamespace := range PRIVILEGED_NAMESPACES {
+				prome2etestrule := e2eCreatePrometheusRule(prometheusName, privilegedNamespace, DUMMY_USER, DUMMY_GROUP, h)
+				err := wait.PollImmediate(5*time.Second, 1*time.Minute, func() (bool, error) {
+					_, err := h.Prometheusop().MonitoringV1().PrometheusRules(privilegedNamespace).Create(context.TODO(), prome2etestrule, metav1.CreateOptions{})
+					if err != nil && errors.IsForbidden(err) {
+						log.Printf("failed creating prometheus rules: %v", err)
+						return true, nil
+					}
+					return false, fmt.Errorf("prometheusrule could be created")
+				})
+				Expect(errors.IsForbidden(err)).To(Equal(true))
+			}
+		}, viper.GetFloat64(config.Tests.PollingTimeout))
+
+		util.GinkgoIt("Privileged users can create prometheusrule in all namespaces", func() {
+			for _, privilegedUser := range PRIVILEGED_USERS {
+				for privilegedNamespace := range PRIVILEGED_NAMESPACES {
+					prome2etestrule := e2eCreatePrometheusRule(prometheusName, privilegedNamespace, privilegedUser, "", h)
+					err := wait.PollImmediate(5*time.Second, 1*time.Minute, func() (bool, error) {
+						_, err := h.Prometheusop().MonitoringV1().PrometheusRules(privilegedNamespace).Create(context.TODO(), prome2etestrule, metav1.CreateOptions{})
+						if err != nil {
+							log.Printf("failed creating prometheus rules: %v", err)
+							return false, nil
+						}
+						return true, nil
+					})
+					Expect(err).NotTo(HaveOccurred())
+				}
+				for _, namespace := range NONPRIV_NAMESPACES {
+					prome2etestrule := e2eCreatePrometheusRule(prometheusName, namespace, privilegedUser, "", h)
+					err := wait.PollImmediate(5*time.Second, 1*time.Minute, func() (bool, error) {
+						_, err := h.Prometheusop().MonitoringV1().PrometheusRules(namespace).Create(context.TODO(), prome2etestrule, metav1.CreateOptions{})
+						if err != nil {
+							log.Printf("failed creating prometheus rules: %v", err)
+							return false, nil
+						}
+						return true, nil
+					})
+					Expect(err).NotTo(HaveOccurred())
+				}
+			}
+		}, viper.GetFloat64(config.Tests.PollingTimeout))
+
+		util.GinkgoIt("Non-privileged users can create prometheusrule all non-privileged namespaces", func() {
+			for _, nonPrivilegedNamespace := range NONPRIV_NAMESPACES {
+				prome2etestrule := e2eCreatePrometheusRule(prometheusName, nonPrivilegedNamespace, DUMMY_USER, "dedicated-admins", h)
+				err := wait.PollImmediate(5*time.Second, 3*time.Minute, func() (bool, error) {
+					_, err := h.Prometheusop().MonitoringV1().PrometheusRules(nonPrivilegedNamespace).Create(context.TODO(), prome2etestrule, metav1.CreateOptions{})
+					if err != nil && errors.IsForbidden(err) {
+						log.Printf("failed creating prometheus rules: %v", err)
+						return false, nil
+					}
+					return true, nil
+				})
```

Comment:
all of the polling is the same except for the namespace you poll, can we please move this inside of the `e2eCreatePrometheusRule` function and just return an error?

Reply:
sorry, just noticed the differences in the error handling of the creation, disregard this comment!